landlab/components/snow/snow_degree_day.py

```diff
@@ -0,0 +1,300 @@
+        },  # T_air
+        "atmosphere_water__precipitation_leq-volume_flux": {
+            "dtype": float,
+            "intent": "in",
+            "optional": False,
+            "units": "m/s",
+            "mapping": "node",
+            "doc": "precipitation rate (in water equivalent)",
+        },  # P
+        "snowpack__degree-day_coefficient": {
+            "dtype": float,
+            "intent": "in",
+            "optional": True,
+            "units": "mm day-1 K-1",
+            "mapping": "node",
+            "doc": "degree-day coefficient",
+        },  # c0
+        "snowpack__degree-day_threshold_temperature": {
+            "dtype": float,
+            "intent": "in",
+            "optional": True,
+            "units": "deg_C",
+            "mapping": "node",
+            "doc": "degree-day threshold temperature to start melt",
+        },  # T0
+        "snowpack__liquid-equivalent_depth": {
+            "dtype": float,
+            "intent": "inout",
+            "optional": True,
+            "units": "m",
+            "mapping": "node",
+            "doc": "snow water equivalent depth",
+        },  # h_swe
+        "snowpack__z_mean_of_mass-per-volume_density": {
+            "dtype": float,
+            "intent": "in",
+            "optional": True,
+            "units": "kg m-3",
+            "mapping": "node",
+            "doc": "snow density",
+        },  # rho_snow
+        # output fields
+        "snowpack__depth": {
+            "dtype": float,
+            "intent": "out",
+            "optional": False,
+            "units": "m",
+            "mapping": "node",
+            "doc": "snow depth",
+        },
+        "snowpack__melt_volume_flux": {
+            "dtype": float,
+            "intent": "out",
+            "optional": False,
+            "units": "m/s",
+            "mapping": "node",
+            "doc": "snow melt volume flux",
+        },
+        "atmosphere_water__time_integral_snowfall_leq-volume_flux": {
+            "dtype": float,
+            "intent": "out",
+            "optional": False,
+            "units": "m",
+            "mapping": "node",
+            "doc": "total precipitation as snow (in water equivalent) during model time",
+        },  # change for landlab (add new variable)
+        "snowpack__time_integral_melt_volume_flux": {
+            "dtype": float,
+            "intent": "out",
+            "optional": False,
+            "units": "m",
+            "mapping": "node",
+            "doc": "total snow melt (in water equivalent) during model time",
+        },  # change for landlab (add new variable)
+    }
+
+    def __init__(self, grid, rho_H2O=1000, T_rain_snow=1, grid_area=0):
+        """Initialize SnowDegreeDay component"""
+
+        super().__init__(grid)
+
+        # parameters
+        self._rho_H2O = rho_H2O
+        self._T_rain_snow = T_rain_snow
+        self._grid_area = grid_area if grid_area > 0 else self._grid.dy * self._grid.dx
```

Comment:
As @mcflugen mentioned, if you need this parameter, make it general enough to accommodate different grid types. There's a grid property grid.area_of_cell that contains the surface area of each grid cell (which might be uniform, as in raster and hex grids, or might be variable, as in VoronoiDelaunay grids)